### PR TITLE
WCAT-354: fix node counting

### PR DESCRIPTION
**Pull request by dagguh**

### CODE REVIEW

**Review by mgrzaslewicz — DISMISSED** [review]

File: src/test/kotlin/com/atlassian/performance/tools/virtualusers/api/EntryPointIT.kt [hunk 1]
```diff
@@ -48,16 +49,19 @@ class EntryPointIT {
...
+            assertThat(actions.map { it.label }.toSet()).containsOnly("Log In", "See System Info", "Set Up")
+            assertThat(actions).haveAtLeast(3, isOk())
+            assertThat(tasks.map { it.label })
+                .contains(ACTING.label, THROTTLING.label, DIAGNOSING.label)
+                .doesNotContain(MYSTERY.label)
+            assertThat(unaccountedTime).isLessThan(Duration.ofSeconds(5))
+            assertThat(nodeResult.nodeDistribution.parent.fileName.toString()).isEqualTo("test-results")
+            assertThat(nodeResult.countVusPerNode())
+                .hasSize(1)
+                .containsValue(2)
```
Comment (dagguh) [inline src/test/kotlin/com/atlassian/performance/tools/virtualusers/api/EntryPointIT.kt:62]: On the TDD:red commit I've got the repro locally:
```
1) 
Expected size:<1> but was:<2> in:
<{"(v7.2.0#72002-sha1:36e1562)"=1, "unknown"=1}>
at EntryPointIT.shouldProduceBrowserLoadMetrics(EntryPointIT.java:60)
at EntryPointIT.shouldProduceBrowserLoadMetrics(EntryPointIT.java:60)
2) 
Expecting:
  <{"(v7.2.0#72002-sha1:36e1562)"=1, "unknown"=1}>
to contain value:
  <2>
at EntryPointIT.shouldProduceBrowserLoadMetrics(EntryPointIT.java:60)
at EntryPointIT.shouldProduceBrowserLoadMetrics(EntryPointIT.java:60)
```

**Comment by dagguh** [comment]
CI failed with:
```
    1) 
    Expecting:
      <["Log In", "See System Info"]>
    to contain only:
      <["Log In", "See System Info", "Set Up"]>
    but could not find the following elements:
      <["Set Up"]>

    at EntryPointIT.shouldProduceBrowserLoadMetrics(EntryPointIT.java:53)
    2) 
    Expecting:
      <{"(v7.2.0#72002-sha1:36e1562)"=1}>
    to contain value:
      <2>
```
And all the diagnoses show:
screenshot

**Comment by dagguh** [comment]
Hmm, now locally I got the same error as on CI. Perhaps the rebase/squash broke it

File: src/main/kotlin/com/atlassian/performance/tools/virtualusers/api/load/ScenarioLoadProcess.kt [hunk 2]
```diff
@@ -41,12 +41,8 @@ class ScenarioLoadProcess : LoadProcess {
         val behavior = options.behavior
         val scenario = behavior.scenario.getConstructor().newInstance()
         val browser = behavior.browser.getConstructor().newInstance()
-        val counter = ClusterNodeCounter()
```
Comment (dagguh) [inline src/main/kotlin/com/atlassian/performance/tools/virtualusers/api/load/ScenarioLoadProcess.kt:44]: Ah yes, previously the counter was done once per process, now my "refactor" moved it to once per thread, that's why it cannot accumulate count above 1.

Comment (dagguh) [reply src/main/kotlin/com/atlassian/performance/tools/virtualusers/api/load/ScenarioLoadProcess.kt]: Fixed.
The screenshot is a red herring

**Comment by dagguh** [comment]
The screenshot is indeed a red herring, because the `See System Info` actions do fail, but our assertions don't care about that.

**Review by mgrzaslewicz — APPROVED** [review]

**Pull request merged**